x-pack/plugins/security_solution/docs/testing/test_plans/detection_response/prebuilt_rules/installation_and_upgrade.md

```diff
@@ -815,6 +821,122 @@ And the Investigation Guide tab should NOT be displayed
 
 TODO: add scenarios https://github.com/elastic/kibana/issues/166215
 
+### Rule upgrade workflow: simple diff algorithm
+
+#### **Scenario: Rule field doesn't have an update and has no custom value**
+
+**Automation** 1 integration test
+
+```Gherkin
+Given at least 1 prebuilt rule is installed in Kibana
+And for this rule there is a new version available
+And the current version of <field> is unchanged from the base version
+And there is no update for <field> in this upgrade
+Then the merged output should be the current version of <field>
+And there should be no conflict
+
+Examples:
+| field      | base_version | current_version | target_version |
+| name       | "A"          | "A"             | "A"            |
+| risk_score | 1            | 1               | 1              |
+```
+
+#### **Scenario: Rule field doesn't have an update but has a custom value**
+
+**Automation** 1 integration test
+
+```Gherkin
+Given at least 1 prebuilt rule is installed in Kibana
+And for this rule there is a new version available
+And the current version of <field> has changed from the base version
+And there is no update for <field> in this upgrade
+Then the merged output should be the current version of <field>
+And there should be no conflict
+
+Examples:
+| field      | base_version | current_version | target_version |
+| name       | "A"          | "B"             | "A"            |
+| risk_score | 1            | 2               | 1              |
+```
+
+#### **Scenario: Rule field has an update and doesn't have a custom value**
+
+**Automation** 1 integration test
+
+```Gherkin
+Given at least 1 prebuilt rule is installed in Kibana
+And for this rule there is a new version available
+And the current version of <field> is unchanged from the base version
+And there is a update for <field> in this upgrade
+Then the merged output should be the target version of <field>
+And there should be no conflict
+
+Examples:
+| field      | base_version | current_version | target_version |
+| name       | "A"          | "A"             | "B"            |
+| risk_score | 1            | 1               | 2              |
+```
+
+#### **Scenario: Rule field has an update and a custom value that are the same**
+
+**Automation** 1 integration test
+
+```Gherkin
+Given at least 1 prebuilt rule is installed in Kibana
+And for this rule there is a new version available
+And the current version of <field> has changed from the base version
+And there is a update for <field> in this upgrade
+And the current version and the update have the same value
+Then the merged output should be the current version of <field>
+And there should be no conflict
+
+CASE: should work the same if rule's base version exists or not
```

Comment:
I'd suggest to extract this into a separate scenario and put it after the `Scenario: Rule field has an update and a custom value that are NOT the same and the rule base version doesn't exist`. I think it feels closer to that one. Because otherwise, why do we have this case here and not in the `Scenario: Rule field doesn't have an update and has no custom value`? When the base version is missing, we can't know if it's a `_AA` (no changes) or a `_BB` (same changes) case.